craft_application/remote/errors.py

```diff
@@ -14,63 +14,41 @@
 
 """Remote build errors."""
 
-from dataclasses import dataclass
+import craft_cli.errors
 
+from craft_application.util import humanize_list
 
-@dataclass(repr=True)
-class RemoteBuildError(Exception):
-    """Unexpected remote build error.
 
-    :param brief: Brief description of error.
-    :param details: Detailed information.
-    """
-
-    brief: str
-    details: str | None = None
-
-    def __str__(self) -> str:
-        """Return the string representation of the error."""
-        components = [self.brief]
-
-        if self.details:
-            components.append(self.details)
-
-        return "\n".join(components)
+class RemoteBuildError(craft_cli.errors.CraftError):
+    """Error for remote builds."""
 
 
 class RemoteBuildGitError(RemoteBuildError):
-    """Git repository cannot be prepared correctly."""
+    """Git repository cannot be prepared correctly.
 
-    def __init__(self, message: str) -> None:
-        self.message = message
-        brief = "Git operation failed"
-        details = message
+    :param message: Git error message.
+    """
 
-        super().__init__(brief=brief, details=details)
+    def __init__(self, message: str) -> None:
+        message = f"Git operation failed with: {message}"
+        super().__init__(message=message)
 
 
 class UnsupportedArchitectureError(RemoteBuildError):
-    """Unsupported architecture error."""
+    """Unsupported architecture error.
+
+    :param architectures: List of unsupported architectures.
+    """
 
     def __init__(self, architectures: list[str]) -> None:
-        brief = "Architecture not supported by the remote builder."
-        details = (
+        message = (
             "The following architectures are not supported by the remote builder: "
-            f"{architectures}.\nPlease remove them from the "
-            "architecture list and try again."
+            f"{humanize_list(architectures, 'and')}."
         )
+        resolution = "Remove them from the architecture list and try again."
 
-        super().__init__(brief=brief, details=details)
+        super().__init__(message=message, resolution=resolution)
 
 
 class RemoteBuildInvalidGitRepoError(RemoteBuildError):
```

Comment:
This may be out of scope for this PR, but should `RemoteBuildInvalidGitRepoError` be a subclass of `RemoteBuildGitError`?

Reply:
No, because `RemoteBuildGitError` is for errors that happened with git, whereas this is saying "this isn't git-able"